junifer/datagrabber/pattern_datalad.py

```diff
@@ -23,122 +25,24 @@ class PatternDataladDataGrabber(DataladDataGrabber, PatternDataGrabber):
-
-        * ``"BOLD"`` :
-
-          .. code-block:: none
-
-            {
-              "mandatory": ["pattern", "space"],
-              "optional": {
-                  "mask": {
-                      "mandatory": ["pattern", "space"],
-                      "optional": []
-                  }
-                  "confounds": {
-                      "mandatory": ["pattern", "format"],
-                      "optional": []
-                  }
-              }
-            }
-
-        * ``"Warp"`` :
-
-          .. code-block:: none
-
-            {
-              "mandatory": ["pattern", "src", "dst"],
-              "optional": []
-            }
-
-        * ``"VBM_GM"`` :
-
-          .. code-block:: none
-
-            {
-              "mandatory": ["pattern", "space"],
-              "optional": []
-            }
-
-        * ``"VBM_WM"`` :
-
-          .. code-block:: none
-
-            {
-              "mandatory": ["pattern", "space"],
-              "optional": []
-            }
-
-        Basically, for each data type, one needs to provide ``mandatory`` keys
-        and can choose to also provide ``optional`` keys. The value for each
-        key is a string. So, one needs to provide necessary data types as a
-        dictionary, for example:
-
-        .. code-block:: none
-
-          {
-              "BOLD": {
-                "pattern": "...",
-                "space": "...",
-              },
-              "T1w": {
-                "pattern": "...",
-                "space": "...",
-              },
-              "Warp": {
-                "pattern": "...",
-                "src": "...",
-                "dst": "...",
-              }
-          }
-
-        taken from :class:`.HCP1200`.
-    replacements : str or list of str
-        Replacements in the ``pattern`` key of each data type. The value needs
-        to be a list of all possible replacements.
-    confounds_format : {"fmriprep", "adhoc"} or None, optional
-        The format of the confounds for the dataset (default None).
-    datadir : str or pathlib.Path or None, optional
-        That directory where the datalad dataset will be cloned. If None,
-        the datalad dataset will be cloned into a temporary directory
-        (default None).
-    rootdir : str or pathlib.Path, optional
+    uri : pydantic.HttpUrl
+        URI of the datalad sibling.
+    types : list of :enum:`.DataType`
+        The data type(s) to grab.
+    patterns : ``DataGrabberPatterns``
+        The datagrabber patterns. Check :class:`DataTypeSchema` for the schema.
+    replacements : list of str
+        All possible replacements in ``patterns.<data_type>.pattern``.
+    rootdir : pathlib.Path, optional
         The path within the datalad dataset to the root directory
-        (default ".").
-    uri : str or None, optional
-        URI of the datalad sibling (default None).
+        (default Path(".")).
+    confounds_format : :enum:`.ConfoundsFormat` or None, optional
```

Comment:
Having an enum for the confounds format will also restrict someone defining their own format and using their own preprocessor for that format.

Reply:
You'd do as shown here: https://juaml.github.io/junifer/pr-preview/pr-364/extending/data_types.html